### PR TITLE
[SPARK-42044][SQL] Fix incorrect error message for `MUST_AGGREGATE_CORRELATED_SCALAR_SUBQUERY`

### PR DESCRIPTION
**Pull request by itholic**
### What changes were proposed in this pull request?

This PR proposes to fix incorrect error message for `MUST_AGGREGATE_CORRELATED_SCALAR_SUBQUERY`

### Why are the changes needed?

The current error message:

`Correlated scalar subqueries in the GROUP BY clause must also be in the aggregate expressions<treeNode>`

is incorrect since it's not related with group by clause at all.

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

`./build/sbt "sql/testOnly org.apache.spark.sql.SQLQueryTestSuite*"`

### CODE REVIEW

**Comment by itholic** [comment]
Test passed.
cc @srielau @MaxGekk @cloud-fan FYI

**Review by srielau — APPROVED** [review]

**Comment by itholic** [comment]
Also cc @cloud-fan

File: core/src/main/resources/error/error-classes.json [hunk 1]
```diff
@@ -1592,7 +1592,7 @@
       },
       "MUST_AGGREGATE_CORRELATED_SCALAR_SUBQUERY" : {
         "message" : [
-          "Correlated scalar subqueries in the GROUP BY clause must also be in the aggregate expressions<treeNode>"
+          "Correlated scalar subqueries must be aggregated to return at most one row."
```
Comment (cloud-fan) [inline core/src/main/resources/error/error-classes.json:1595]: not related to this PR, but how is this different from `MUST_AGGREGATE_CORRELATED_SCALAR_SUBQUERY_OUTPUT`?

**Review by cloud-fan — APPROVED** [review]

File: core/src/main/resources/error/error-classes.json (hunk 1 above)
Comment (itholic) [reply core/src/main/resources/error/error-classes.json:1595]: Seems like we can integrate it into `MUST_AGGREGATE_CORRELATED_SCALAR_SUBQUERY`.

Let me address it as follow-up soon.

**Comment by HyukjinKwon** [comment]
Merged to master.

**Review by HyukjinKwon — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master.

File: core/src/main/resources/error/error-classes.json (hunk 1 above)
Comment (cloud-fan) [reply core/src/main/resources/error/error-classes.json:1670]: SGTM. Is the ticket created?

Comment (itholic) [reply core/src/main/resources/error/error-classes.json:1670]: Here is the ticket just created: https://issues.apache.org/jira/browse/SPARK-42239

Let me address this soon.

Comment (itholic) [reply core/src/main/resources/error/error-classes.json:1670]: Just opened the PR: https://github.com/apache/spark/pull/39806

**Pull request closed without merging**